api/app/geospatial/wps_dataset.py

```diff
@@ -0,0 +1,202 @@
+from typing import Optional
+from osgeo import gdal, osr
+import numpy as np
+
+from app.utils.geospatial import GDALResamplingMethod
+
+
+class WPSDataset:
+    """
+    A wrapper around gdal datasets for common operations
+    """
+
+    def __init__(self, ds_path: Optional[str], ds=None, band=1, chunk_size=256, access=gdal.GA_ReadOnly, datatype=gdal.GDT_Byte):
+        self.ds = ds
+        self.ds_path = ds_path
+        self.band = band
+        self.chunk_size = chunk_size
+        self.access = access
+        self.datatype = datatype
+
+    def __enter__(self):
+        if self.ds is None:
+            self.ds: gdal.Dataset = gdal.Open(self.ds_path, self.access)
+        return self
+
+    def __exit__(self, *_):
+        self.ds = None
+
+    def __mul__(self, other):
+        """
+        Multiplies this WPSDataset with the other WPSDataset
+
+        :param other: WPSDataset
+        :raises ValueError: Raised if this and other WPSDataset have mismatched raster dimensions
+        :return: a new WPSDataset
+        """
+        # Get raster dimensions
+        x_size = self.ds.RasterXSize
+        y_size = self.ds.RasterYSize
+
+        # Check if the dimensions of both rasters match
+        if x_size != other.ds.RasterXSize or y_size != other.ds.RasterYSize:
```

Comment:
We should probably check that the two datasets have the same spatial reference (and maybe origin?).

Reply:
Done in: https://github.com/bcgov/wps/pull/4010/commits/0c4e81a4577272d8742e604935c5af4bd89e3d69